app/apikey.go

```diff
@@ -122,6 +125,40 @@ func (s *APIKeyClient) deleteAPIKey(
 	return PrintProto(resp)
 }
 
+// a hacky version of time.ParseDuration() which considers days ('d') a valid time unit.
+func parseDuration(s string) (time.Duration, error) {
+	var d time.Duration
+	durationString := s
+	parts := strings.Split(s, "d")
+	if len(parts) == 2 {
+		days, err := strconv.ParseInt(parts[0], 10, 32)
+		if err != nil {
+			return d, fmt.Errorf("time: invalid duration \"%s\"", s)
+		}
+		if days < 0 {
```

Comment:
What happens if the user specifies `0d`?

Reply:
we'd return 0 i believe. technically `0d24h` works too...

Reply:
i added an addition check in the `Action` function to reject durations of 0. wdyt?